oneanddone/tasks/views.py

```diff
@@ -350,19 +351,125 @@ def post(self, *args, **kwargs):
         return redirect('tasks.feedback', attempt.pk)
 
 
-class TaskDetailView(generic.DetailView):
-    model = Task
-    template_name = 'tasks/detail.html'
-    allow_expired_tasks = True
+class TaskAttemptDetailView(LoginRequiredMixin,
+                            MyStaffUserRequiredMixin,
+                            generic.View):
+
+    def get(self, request, *args, **kwargs):
+        view = TaskAttemptDisplayView.as_view()
+        return view(request, *args, **kwargs)
+
+    def post(self, request, *args, **kwargs):
+        view = SubmitAdminVerificationResponseView.as_view()
+        return view(request, *args, **kwargs)
+
+
+class TaskAttemptDisplayView(LoginRequiredMixin,
+                             MyStaffUserRequiredMixin,
+                             generic.DetailView):
+    model = TaskAttempt
+    template_name = 'tasks/attempt_detail.html'
 
     def get_context_data(self, *args, **kwargs):
-        ctx = super(TaskDetailView, self).get_context_data(*args, **kwargs)
+        ctx = super(TaskAttemptDisplayView, self).get_context_data(*args, **kwargs)
+        ctx['form'] = SubmitVerifiedTaskForm()
+        return ctx
+
+
+class SubmitAdminVerificationResponseView(LoginRequiredMixin,
+                                          MyStaffUserRequiredMixin,
+                                          generic.detail.SingleObjectMixin,
+                                          generic.FormView):
+
+    form_class = SubmitVerifiedTaskForm
+    model = TaskAttempt
+    template_name = 'tasks/attempt_detail.html'
+
+    def post(self, request, *args, **kwargs):
+        self.object = self.get_object()
+        return super(SubmitAdminVerificationResponseView, self).post(request, *args, **kwargs)
+
+    def form_valid(self, form):
+
+        attempt = self.get_object()
+
+        communication = form.save(commit=False)
```

Comment:
It could make sense to move the email sending code in the `SubmitVerifiedTaskForm` to keep it DRY and more tight. So when you save the form emails go out etc.